simpl/db/mongodb.py

```diff
@@ -115,6 +115,55 @@ def scrub(data):
         raise ValidationError("Input '%s' not permitted: %s" % (data, exc))
 
 
+def build_text_search(strings, name_field='name'):
```

Comment:
These are all valid use cases, but would go beyond what we have in prod today. So I would request we (you :smile:) add them as issues (feature requests) and not add to this PR.

Note that the current implementation is not mongo-specific on the front end (just one or more strings coming in as `?q=search1,search2&q=search3`). We talked about implementing a github-like query syntax (ex. `no:milestone`). That might conflict with mongo-style negations if we implement them now.